torch_disc/torch_disc/csrc/disc_jit.cpp

```diff
@@ -161,29 +147,95 @@ std::string MhloConversaion(const std::shared_ptr<Graph>& graph) {
   std::ofstream outfile(in_fname);
   outfile << parsable_mlir << std::endl;
   outfile.flush();
-  return in_fname;
+  return std::make_tuple(in_fname, input_dev_str, output_dev_str);
 }
 
-void CallDiscCompiler(const std::string& mlir_fname) {
+std::string CallDiscCompiler(const std::string& mlir_fname) {
   std::string out_fname = mlir_fname + ".out";
   std::string cmd = DiscCMD(mlir_fname, out_fname);
   TORCH_CHECK(std::system(cmd.c_str()) == 0,
               "disc compile failed with cmd: " + cmd);
+  return out_fname;
 }
 
-void DiscCompilation(const std::shared_ptr<Graph>& graph) {
-  for (auto node : graph->nodes()) {
-    if (node->kind() == prim::FusionGroup) {
-      auto sub_graph = node->g(attr::Subgraph);
-      std::string fname = MhloConversaion(sub_graph);
-      CallDiscCompiler(fname);
+std::vector<const torch::jit::Value*> ToConstValue(
+    c10::ArrayRef<torch::jit::Value*> values) {
+  std::vector<const torch::jit::Value*> const_vals;
+  std::copy(values.begin(), values.end(), std::back_inserter(const_vals));
```

Comment:
The function name is confusing since there is nothing about const.

Reply:
This function just cast `vector<T>` to `vector<const T>` to work well with `torch::blade::ShapeTypeSpec::GetShapeTypeSpec`.

BTW, I'm curious that why using `const vector<const T>` as args, it seems `const vector<T>` is enough, c.f. https://stackoverflow.com/a/6955332